<launch>
	<arg name="mode" default="preset" />
	<arg name="enable_depth" default="1" />
	<arg name="enable_color" default="1" />
	<arg name="enable_pointcloud" default="0" />
	<arg name="camera" default="R200" />
	<arg name="COLOR_BACKLIGHT_COMPENSATION" default="1" />
	<arg name="COLOR_BRIGHTNESS" default="56" />
	<arg name="COLOR_CONTRAST" default="32" />
	<arg name="COLOR_GAIN" default="32" />
	<arg name="COLOR_GAMMA" default="220" />
	<arg name="COLOR_HUE" default="0" />
	<arg name="COLOR_SATURATION" default="128" />
	<arg name="COLOR_SHARPNESS" default="0" />
	<arg name="COLOR_ENABLE_AUTO_WHITE_BALANCE" default="0" />
	<arg name="R200_LR_AUTO_EXPOSURE_ENABLED" default="0" />
	<arg name="R200_LR_GAIN" default="400" />
	<arg name="R200_LR_EXPOSURE" default="164" />
	<arg name="R200_EMITTER_ENABLED" default="1" />
	<arg name="R200_DEPTH_UNITS" default="1000" />
	<arg name="R200_DEPTH_CLAMP_MIN" default="0" />
	<arg name="R200_DEPTH_CLAMP_MAX" default="65535" />
	<arg name="R200_DISPARITY_MULTIPLIER" default="32" />
	<arg name="R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_DECREMENT" default="5" />
	<arg name="R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_INCREMENT" default="5" />
	<arg name="R200_DEPTH_CONTROL_MEDIAN_THRESHOLD" default="192" />
	<arg name="R200_DEPTH_CONTROL_SCORE_MINIMUM_THRESHOLD" default="1" />
	<arg name="R200_DEPTH_CONTROL_SCORE_MAXIMUM_THRESHOLD" default="512" />
	<arg name="R200_DEPTH_CONTROL_TEXTURE_COUNT_THRESHOLD" default="6" />
	<arg name="R200_DEPTH_CONTROL_TEXTURE_DIFFERENCE_THRESHOLD" default="24" />
	<arg name="R200_DEPTH_CONTROL_SECOND_PEAK_THRESHOLD" default="27" />
	<arg name="R200_DEPTH_CONTROL_NEIGHBOR_THRESHOLD" default="7" />
	<arg name="R200_DEPTH_CONTROL_LR_THRESHOLD" default="24" />


  	<node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>
  	
	<node pkg="nodelet" type="nodelet" name="RealsenseNodelet"
        args="load realsense_camera/RealsenseNodelet standalone_nodelet 
        mode $(arg mode)
        enable_color $(arg enable_color)
        enable_depth $(arg enable_depth)
        enable_pointcloud $(arg enable_pointcloud)
	COLOR_BACKLIGHT_COMPENSATION $(arg COLOR_BACKLIGHT_COMPENSATION) 
	COLOR_BRIGHTNESS $(arg COLOR_BRIGHTNESS)
	COLOR_CONTRAST $(arg COLOR_CONTRAST)
	COLOR_GAIN $(arg COLOR_GAIN)
	COLOR_GAMMA $(arg COLOR_GAMMA)
	COLOR_HUE $(arg COLOR_HUE)
	COLOR_SATURATION $(arg COLOR_SATURATION)
	COLOR_SHARPNESS $(arg COLOR_SHARPNESS)
	COLOR_ENABLE_AUTO_WHITE_BALANCE  $(arg COLOR_ENABLE_AUTO_WHITE_BALANCE)
	R200_LR_AUTO_EXPOSURE_ENABLED $(arg R200_LR_AUTO_EXPOSURE_ENABLED)
	R200_LR_GAIN $(arg R200_LR_GAIN)
	R200_LR_EXPOSURE $(arg R200_LR_EXPOSURE)
	R200_EMITTER_ENABLED $(arg R200_EMITTER_ENABLED)
	R200_DEPTH_UNITS $(arg R200_DEPTH_UNITS)
	R200_DEPTH_CLAMP_MIN $(arg R200_DEPTH_CLAMP_MIN)
	R200_DEPTH_CLAMP_MAX $(arg R200_DEPTH_CLAMP_MAX)
	R200_DISPARITY_MULTIPLIER $(arg R200_DISPARITY_MULTIPLIER)
	R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_DECREMENT $(arg R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_DECREMENT)
	R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_INCREMENT $(arg R200_DEPTH_CONTROL_ESTIMATE_MEDIAN_INCREMENT)
	R200_DEPTH_CONTROL_MEDIAN_THRESHOLD $(arg R200_DEPTH_CONTROL_MEDIAN_THRESHOLD)
	R200_DEPTH_CONTROL_SCORE_MINIMUM_THRESHOLD $(arg R200_DEPTH_CONTROL_SCORE_MINIMUM_THRESHOLD)
	R200_DEPTH_CONTROL_SCORE_MAXIMUM_THRESHOLD $(arg R200_DEPTH_CONTROL_SCORE_MAXIMUM_THRESHOLD)
	R200_DEPTH_CONTROL_TEXTURE_COUNT_THRESHOLD $(arg R200_DEPTH_CONTROL_TEXTURE_COUNT_THRESHOLD)
	R200_DEPTH_CONTROL_TEXTURE_DIFFERENCE_THRESHOLD $(arg R200_DEPTH_CONTROL_TEXTURE_DIFFERENCE_THRESHOLD)
	R200_DEPTH_CONTROL_SECOND_PEAK_THRESHOLD $(arg R200_DEPTH_CONTROL_SECOND_PEAK_THRESHOLD)
	R200_DEPTH_CONTROL_NEIGHBOR_THRESHOLD $(arg R200_DEPTH_CONTROL_NEIGHBOR_THRESHOLD)
	R200_DEPTH_CONTROL_LR_THRESHOLD $(arg R200_DEPTH_CONTROL_LR_THRESHOLD) ">

  	</node>
  	
</launch> 

